tests/components/mqtt/conftest.py

```diff
@@ -39,15 +42,31 @@ def temp_dir_prefix() -> str:
 
 
 @pytest.fixture(autouse=True)
-def mock_temp_dir(temp_dir_prefix: str) -> Generator[str]:
+async def mock_temp_dir(
+    hass: HomeAssistant, tmp_path: Path, temp_dir_prefix: str
+) -> AsyncGenerator[str]:
     """Mock the certificate temp directory."""
-    with patch(
-        # Patch temp dir name to avoid tests fail running in parallel
-        "homeassistant.components.mqtt.util.TEMP_DIR_NAME",
-        f"home-assistant-mqtt-{temp_dir_prefix}-{getrandbits(10):03x}",
-    ) as mocked_temp_dir:
+    mqtt_temp_dir = f"home-assistant-mqtt-{temp_dir_prefix}-{getrandbits(10):03x}"
+    with (
+        patch(
+            "homeassistant.components.mqtt.util.tempfile.gettempdir",
+            return_value=tmp_path,
+        ),
+        patch(
+            # Patch temp dir name to avoid tests fail running in parallel
+            "homeassistant.components.mqtt.util.TEMP_DIR_NAME",
+            mqtt_temp_dir,
+        ) as mocked_temp_dir,
+    ):
         yield mocked_temp_dir
 
+    # Clean up temp files after test
+    def _cleanup() -> None:
+        with contextlib.suppress(OSError):
+            shutil.rmtree(tmp_path)
+
+    await hass.async_add_executor_job(_cleanup)
```

Comment:
```suggestion
    with contextlib.suppress(OSError):
        await hass.async_add_executor_job(shutil.rmtree, tmp_path)
```

Reply:
Although I'm not 100% sure this is needed as `tmp_path` should get deleted by the fixture after the test

Reply:
Right, I think patching `gettenpfile` is probably enough. Let's remove the clean up part for now.

Reply:
Corrected the PR's description, removed cleanup